vault/identity_lookup_test.go

```diff
@@ -45,6 +45,19 @@ func TestIdentityStore_Lookup_Entity(t *testing.T) {
 		t.Fatal(err)
 	}
 
+	if len(entity.Aliases) != 1 {
```

Comment:
This test addition is unrelated to entity merges, but this would have failed before my change to `MemDBEntityByID`, so I'm adding it here to make sure this behaviour is tested explicitly.